### PR TITLE
Feature/confirm delete dialog by clicking only delete button

### PR DESCRIPTION
**Pull request by vamsiyeluriGD**
### Description
Added dialog check while deleting a product. Made sure that product is only getting deleted on click on delete button.

### Screen recording:
Link: [Feature/ confirm delete dialog by clicking only delete button](https://drive.google.com/file/d/1VrIovgqujHjg-kIpsldDUluokF1PIBkG/view?usp=share_link)

### Issue
Fixes: #32 

###Referenced PR : [Feature/confirm delete dialog ] (https://github.com/vnazarchukGD/angular-time-killer/pull/36)

### CODE REVIEW

File: src/app/products/components/product-delete-confirm-dialog/product-delete-confirm-dialog.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+import { ProductDeleteConfirmDialogComponent } from "./product-delete-confirm-dialog.component";
+import { ProductDeleteConfirmDialogHarness } from "./product-delete-confirm-dialog.harness";
+
+const product = {
+  name: "test product",
+  description: "test description",
+  picture: "picture url"
+}
+
+describe("AppModule => ProductDeleteConfirmDialogComponent", () => {
```
Comment (ooow) [inline src/app/products/components/product-delete-confirm-dialog/product-delete-confirm-dialog.component.spec.ts:21]: Leave just the component name.

File: src/app/products/components/product-delete-confirm-dialog/product-delete-confirm-dialog.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+    component.openDialog();
+    fixture.detectChanges();
+    await fixture.whenStable();
+
+    dialogHarness = await rootLoader.getHarness(ProductDeleteConfirmDialogHarness);
+  })
+
+  it("should be created", () => {
+    expect(component).not.toBeNull();
```
Comment (ooow) [inline src/app/products/components/product-delete-confirm-dialog/product-delete-confirm-dialog.component.spec.ts:73]: This test is not needed, all other tests bellow cover this.

File: src/app/products/components/product-delete-confirm-dialog/product-delete-confirm-dialog.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+  });
+
+  it("should be rendered appropriately", async () => {
```
Comment (ooow) [inline src/app/products/components/product-delete-confirm-dialog/product-delete-confirm-dialog.component.spec.ts:76]: should be - is extra, please cut repeating parts to make the description more compact but still informative. E.g. 
'it should be rendered appropriately' -> 'it renders component'

File: src/app/products/components/product-delete-confirm-dialog/product-delete-confirm-dialog.component.ts [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import { Component, Inject } from '@angular/core';
+import { MatDialogRef, MAT_DIALOG_DATA } from '@angular/material/dialog';
+import { ProductDeleteConfirmation } from 'src/app/shared/models/product-delete-confirmation';
+@Component({
```
Comment (ooow) [inline src/app/products/components/product-delete-confirm-dialog/product-delete-confirm-dialog.component.ts:4]: Add an empty line after imports.

**Review by ooow — APPROVED** [review]
LGTM. Please fix comments before submitting.

File: src/app/products/components/product-delete-confirm-dialog/product-delete-confirm-dialog.component.spec.ts (hunk 1 above)
Comment (vamsiyeluriGD) [reply src/app/products/components/product-delete-confirm-dialog/product-delete-confirm-dialog.component.spec.ts]: Mentioned change is completed hence resolving this conversation.

**Comment by vamsiyeluriGD** [comment]
Mentioned changes are completed hence resolving these conversations and merging this PR.

**Pull request merged**